### PR TITLE
rid join

### PR DESCRIPTION
**Pull request by xiaoxiao921**

### CODE REVIEW

File: BigBaseV2/src/views/network/view_session.cpp [hunk 1]
```diff
@@ -1,17 +1,64 @@
...
+		if (g_pointers->m_start_get_session_by_gamer_handle(0, &player_handle, 1, &result, 1, &success, &state))
+		{
+			while (state == 1)
+				script::get_current()->yield();
+
+			if (state == 3 && success)
+			{
+				g->session.join_queued = true;
+				g->session.info = result.m_session_info;
+				big::session::join_type({ eSessionType::NEW_PUBLIC });
```
Comment (Yimura) [inline BigBaseV2/src/views/network/view_session.cpp]: Why prepend big namespace?

File: BigBaseV2/src/views/network/view_session.cpp [hunk 2]
```diff
@@ -1,17 +1,64 @@
...
+				if (SCRIPT::GET_NUMBER_OF_THREADS_RUNNING_THE_SCRIPT_WITH_THIS_HASH(RAGE_JOAAT("maintransition")) == 0)
+				{
+					g->session.join_queued = false;
+					notify::above_map("RID join failed, unable to launch maintransition");
```
Comment (Yimura) [inline BigBaseV2/src/views/network/view_session.cpp]: Might get obstructed by the player list rendering over this, prefer using the notification service.

File: BigBaseV2/src/views/network/view_session.cpp [hunk 3]
```diff
@@ -1,17 +1,64 @@
...
+#include "rage/rlSessionByGamerTaskResult.hpp"
 
 namespace big
 {
+	static void join_by_rockstar_id(uint64_t rid)
+	{
+		if (SCRIPT::GET_NUMBER_OF_THREADS_RUNNING_THE_SCRIPT_WITH_THIS_HASH(RAGE_JOAAT("maintransition")) != 0 ||
+			STREAMING::IS_PLAYER_SWITCH_IN_PROGRESS())
+		{
+			notify::above_map("Cannot RID join now");
```
Comment (Yimura) [inline BigBaseV2/src/views/network/view_session.cpp]: Might get obstructed by the player list rendering over this, prefer using the notification service.

File: BigBaseV2/src/views/network/view_session.cpp [hunk 4]
```diff
@@ -1,17 +1,64 @@
...
+			}
+		}
+
+		notify::above_map("RID join failed");
+	}
+
 	void view::session()
 	{
+		static char rid_text[13] = "";
+		ImGui::InputText("Input RID", rid_text, sizeof(rid_text));
```
Comment (Yimura) [inline BigBaseV2/src/views/network/view_session.cpp]: InputScalar?

**Review by Yimura — CHANGES_REQUESTED** [review]

**Review by Yimura — DISMISSED** [review]

File: BigBaseV2/src/views/network/view_session.cpp [hunk 5]
```diff
@@ -1,17 +1,63 @@
...
 #include "util/session.hpp"
 #include "views/view.hpp"
+#include "pointers.hpp"
+#include "gta/joaat.hpp"
+#include "util/notify.hpp"
+#include "rage/rlSessionByGamerTaskResult.hpp"
 
 namespace big
 {
+	static void join_by_rockstar_id(uint64_t rid)
```
Comment (tupoy-ya) [inline BigBaseV2/src/views/network/view_session.cpp]: Why not put it in `util/session.hpp`?

**Review by Yimura — APPROVED** [review]

**Pull request merged**